src/components/Feed/Feed.js

```diff
@@ -3,12 +3,42 @@ import { Route } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
 import './Feed.css';
-
-// import Post from '../../components/Post/Post';
+import Post from '../../components/Post/Post';
+import { user, posts } from '../../Data/mockData';
 
 const Feed = () => {
+  const [currentUserPosts, setCurrentUserPosts] = useState([])
+  const [allPosts, setAllPosts] = useState([])
+
+  useEffect(() => {
+    setCurrentUserPosts(user.posts)
+    setAllPosts(posts)
+    // console.log('currentUserPosts', currentUserPosts)
+    // console.log('allPosts', allPosts)
+  }, [currentUserPosts, allPosts])
+
+  const displayPostsForCurrentUser = (postsToDisplay) => {
+    return postsToDisplay.map(post => {
+      return (
+        <Post
+          userName={post.userName}
+          title={post.title}
+          date={post.date}
+          location={post.location}
+          description={post.description}
+          distance={post.distance}
+          imageUrl={post.image}
+        />
+      )
+    })
+  }
+
   return (
-    <h2>Feed goes here</h2>
+    <div className='post-container'>
+      {/* Conditionally render the feed to only display all posts on the home page
+        and the current user's posts on the profile page */}
+      {displayPostsForCurrentUser(currentUserPosts)}
```

Comment:
Maybe you've changed this - I'm not seeing how we can use both the currentUserPosts and allPosts. Did you figure this out in the updated files?

Reply:
Take a look at the most recent commit `Conditionally render feed`.